api/src/main/java/io/minio/S3Base.java

```diff
@@ -193,6 +197,7 @@ protected S3Base(S3Base client) {
     this.region = client.region;
     this.provider = client.provider;
     this.httpClient = client.httpClient;
+    this.closeHttpClient = false;
```

Comment:
```suggestion
    this.closeHttpClient = client.closeHttpClient;
```

Reply:
That depends on the purpose of this "copy" constructor, i.e. what semantics we want,
If the caller may use both instances of S3Base, only one should close the HTTP client (and then the other is not usable anymore).
If the caller is supposed to use only the new instance of S3Base, yes, we should copy the flag.

For now, I will copy the flag.